mtest/libmscore/albumsIO/tst_albumsIO.cpp

```diff
@@ -0,0 +1,285 @@
+//    }
+    QVERIFY(compareFilesFromPaths(root + "/" + DIR + "imported" + QDir::separator() + QString(file) + ".msca",
+                                  fi.absolutePath() + QDir::separator() + "imported" + QDir::separator() + QString(file) + "_generated"
+                                  + ".msca"));
+}
+
+//---------------------------------------------------------
+//   stringsTest
+//---------------------------------------------------------
+
+void TestAlbumsIO::stringsTest(const char* file)
+{
+    MScore::debugMode = true;
+    Album* album = readAlbum(DIR + QString(file) + ".msca");
+    QVERIFY(album);
+
+    // load scores
+    loadScores(album);
+
+    std::cout << "Loading completed..." << std::endl;
+
+    auto x = album->composers();
+    QVERIFY(x.size() == 2);
+    QCOMPARE(x.at(0), QString("Sergios - Anestis Kefalidis"));
+    QCOMPARE(x.at(1), QString("Oregano"));
+    auto y = album->lyricists();
+    QVERIFY(y.size() == 1);
+    QCOMPARE(y.at(0), QString("Garlic"));
+    auto z = album->scoreTitles();
+    QVERIFY(z.size() == 3);
+    QCOMPARE(z.at(0), QString("Piano1"));
+    QCOMPARE(z.at(1), QString("Piano2"));
+    QCOMPARE(z.at(2), QString("Piano3"));
+
+    album->removeScore(0);
+
+    auto x2 = album->composers();
+    QVERIFY(x2.size() == 1);
+    QCOMPARE(x2.at(0), QString("Oregano"));
+    auto y2 = album->lyricists();
+    QVERIFY(y2.size() == 1);
+    QCOMPARE(y2.at(0), QString("Garlic"));
+    auto z2 = album->scoreTitles();
+    QVERIFY(z2.size() == 2);
+    QCOMPARE(z2.at(0), QString("Piano2"));
+    QCOMPARE(z2.at(1), QString("Piano3"));
+
+    delete album;
+}
+
+//---------------------------------------------------------
+//   addRemoveTest
+//---------------------------------------------------------
+
+void TestAlbumsIO::addRemoveTest(const char* file)
+{
+    MScore::debugMode = true;
+    Album* album = readAlbum(DIR + QString(file) + ".msca");
+    QVERIFY(album);
+
+    // load scores
+    loadScores(album);
+
+    if (strcmp(file, "smallPianoAlbum") == 0) {
+        QVERIFY(album->albumItems().size() == 3);
+    } else if (strcmp(file, "albumWithParts") == 0) {
+        QVERIFY(album->albumItems().size() == 2);
+    }
+
+    int albumSize = album->albumItems().size();
+    int lastIndex = albumSize - 1;
+
+    MasterScore* ms = album->albumItems().at(lastIndex)->score;
+    album->removeScore(ms);
+    QCOMPARE(int(album->albumItems().size()), albumSize - 1);
+    album->addScore(ms);
+    QCOMPARE(int(album->albumItems().size()), albumSize);
+
+    album->createDominant();
+    MasterScore* dominant = album->getDominant();
+    for (auto& e : dominant->excerpts()) {
+        QCOMPARE(e->partScore()->isMaster(), true);
+    }
+    auto excerptCheck = [&]() {
+                            for (auto& e : dominant->excerpts()) {
+                                QCOMPARE(int(e->partScore()->movements()->size()), int(dominant->movements()->size()));
+                            }
+                        };
```

Comment:
You can avoid the massive indentation by declaring this lambda expression as a separate function:

```c++
static void excerptCheck(MasterScore* dominant)
{
    for (auto& e : dominant->excerpts()) {
        QCOMPARE(int(e->partScore()->movements()->size()), int(dominant->movements()->size()));
    }
}
```

Declaring it `static` means it won't (and can't) be used anywhere except this file, so the compiler is free to inline it and doesn't need to keep track of it for external linkage. In other words, the end result is the same as the lambda expression but the code is neater and Uncrustify doesn't complain.

If you go down this route then you should put the static function at the very top of the file because you can't use it before it is declared.

P.S. You might want to call it `dominantScore` rather than just `dominant` for the sake of people who haven't seen it before.